src/Build/Graph/ProjectGraph.cs

```diff
@@ -75,6 +76,16 @@ public sealed class ProjectGraph
 
         public IReadOnlyCollection<ProjectGraphNode> GraphRoots { get; }
 
+        /// <summary>
+        ///     Gets a distinct enumeration of all of the environment variables reads during evaluation.
+        /// </summary>
+        public IEnumerable<string> EnvironmentVariableReads => _projectNodesTopologicallySorted?.Value.Select(p => p.ProjectInstance).SelectMany(pi => pi.EnvironmentVariableReads).Distinct();
+
+        /// <summary>
+        ///     Get a distinct enumeration of all the uninitialized property reads during evaluation.
+        /// </summary>
+        public IEnumerable<string> UninitializedPropertyReads => _projectNodesTopologicallySorted?.Value.Select(p => p.ProjectInstance).SelectMany(pi => pi.UninitializedPropertyReads).Distinct();
```

Comment:
>UninitializedPropertyReads [](start = 35, length = 26)

At least at the static api level, I would rename this to something like 'AbsentEnvironmentVariableProbes'.
Uninitialized properties, even though represent the same concept, involves a couple of hops in the semantic that MSBuild gives to property/env var interactions, and looks somewhat asymmetrical to 'EnvironmentVariableReads'